### PR TITLE
docs(next-release): Add docs for i18n migration

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes

- [x] Fix typos I had in my previous changesets
- [x] Add documentation
- [x] Make migration guide formatting consistent

#### Description of changes
Note that #2642 needs to be merged first, because I'm using its migration guide formatting. Tests will run once #2642 is merged and I change the target branch.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx [hunk 1]
```diff
@@ -25,9 +25,8 @@ yarn add aws-amplify @aws-amplify/ui-angular@3.x
 
 The following breaking changes were introduced in `@aws-amplify/ui-angular` major version `3.0`:
 
-#### Automatic signin on signup logic moved to `aws-amplify`. 
 
-##### 1. Update the handleSignUp function override call
+#### 1. Update the handleSignUp function override call
```
Comment (wlee221) [inline docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx:29]: We were using flat numbered list in React migration guide, so adjusting Angular guide formatting as so.

File: .changeset/witty-seas-meow.md [hunk 2]
```diff
@@ -1,9 +1,8 @@
 ---
-'@aws-amplify/ui-vue': major
+'@aws-amplify/ui-react': major
```
Comment (wlee221) [inline .changeset/witty-seas-meow.md:2]: I accidentally two changesets for `vue` but not for `react` 🤦

File: docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx [hunk 3]
```diff
@@ -25,9 +25,8 @@ yarn add aws-amplify @aws-amplify/ui-angular@3.x
 
 The following breaking changes were introduced in `@aws-amplify/ui-angular` major version `3.0`:
 
-#### Automatic signin on signup logic moved to `aws-amplify`.
```
Comment (reesscot) [inline docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx:28]: Why is this removed? The two steps below are related subitems of this. Is there a better way to say this?

File: docs/src/pages/[platform]/getting-started/migration/migration.react.mdx [hunk 4]
```diff
@@ -79,9 +79,7 @@ Hopefully this change won't affect your code but will allow for more customizati
   The TextAreaField component does apply 1 style prop directly on the `textarea` element: `resize`. We felt this one makes sense to apply direclty on the the `textarea` element and not the wrapper element.
 </Alert>
 
-#### Automatic signin on signup logic moved to `aws-amplify`.
```
Comment (reesscot) [inline docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:82]: Same comment as above

File: docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx (hunk 3 above)
Comment (wlee221) [reply docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx:28]: Oh yeah, that was my attempt to flatten this list. 

My thought was that "Automatic signin on signup logic moved to `aws-amplify`" is more of an internal change. We have always done `autoSignIn` from customer's POV. That's why I went ahead and removed it, but what do you think?

**Review by 0618 — APPROVED** [review]

**Review by ioanabrooks — DISMISSED** [review]

**Review by ioanabrooks — APPROVED** [review]

**Review by 0618 — APPROVED** [review]

**Pull request merged**